docs/tutorials/clojure-guide.md

```diff
@@ -0,0 +1,238 @@
+official [lsp-mode site](https://emacs-lsp.github.io/lsp-mode/page/main-features/).
+
+### Syntax check
+
+It checks for errors and warnings on your code while you write it, it uses [clj-kondo](https://github.com/clj-kondo/clj-kondo), 
+another amazing library that lints Clojure code, in the back-end so it's not required to install `clj-kondo` since `clojure-lsp` will
+use it automatically.
+
+It will also check for public unused functions/variables which is something that, as far as I know, does not exist in 
+any other Clojure library/package:
+
+![syntax-checker](images/clojure-syntax-checker.png "syntax-checker")
+
+### Find definition/references
+
+You can find a function definition with `lsp-find-definition` or find all the references to that 
+symbol, variable or function with `lsp-find-references` or `lsp-ui-peek-find-references` if 
+[lsp-ui](https://emacs-lsp.github.io/lsp-ui/) is available. 
+
+![find-references](images/clojure-find-references.gif "find-references")
+
+<details>
+<summary>Doom emacs users</summary>
+
+You may need to remove the lookup handlers conflict from `cider` and/or `clj-refactor` if you want to use this LSP feature.
+
+```elisp
+(use-package! cider
+  :after clojure-mode
+  :config
+  (set-lookup-handlers! 'cider-mode nil))
+  
+(use-package! clj-refactor
+  :after clojure-mode
+  :config
+  (set-lookup-handlers! 'clj-refactor-mode nil))
+```
+
+</details>
+
+### Completion
+
+By default, `lsp-mode` uses `company-mode` as its completion front-end. When
+present, `company-mode` and enabled for `lsp-mode`, it will be auto-configured and it will just work using the completion items returned by the LSP server. 
+If you do not want LSP completion, It's possible to use `cider` completion instead with:
+
+```elisp
+(setq lsp-enable-completion-at-point nil) ; use cider completion
+```
+
+![completion](images/clojure-completion.png "completion")
+
+See also [CIDER code completion](https://docs.cider.mx/cider/usage/code_completion.html)
+
+### Code lens
+
+LSP supports code lenses, actionable text snippets that a server may want to display in a client IDE. `clojure-lsp` 
+uses it to show the reference count of a function/variable. They are disabled by default so you need to manually enable them via:
+
+```elisp
+(setq lsp-lens-enable t)
+```
+
+![reference code-lens](images/clojure-code-lens.png "reference code-lens")
+
+### Call hierarchy
+
+This feature returns the call hierarchy of a function/method.
+As of now, `clojure-lsp` only supports returning the incoming-call hierarchy.
+
+![incoming-call hierarchy](images/clojure-call-hierarchy.png "incoming-call hierarchy")
+
+It uses [lsp-treemacs](https://github.com/emacs-lsp/lsp-treemacs), which you should install to use this feature.
+
+### Formatting
+
+For LSP formatting, `clojure-lsp` use [cljfmt](https://github.com/weavejester/cljfmt) with support for all known `cljfmt` 
```

Comment:
How exactly do you use it? Are you shelling out to it on save or on `indent-line`? I'm asking because CIDER also uses it, but not by default, as it might conflict with the indentation you'd get from Emacs's (clojure-mode's) indentation engine.

Reply:
It's a common lsp command `lsp-format-buffer` or `lsp-format-region` so it may not conflict with CIDER, but I'll this command mention, thanks

Reply:
If it's something you have to invoke explicitly - yeah, it won't conflict with CIDER/clojure-mode.